### PR TITLE
SPMI: Include example diffs in markdown summary

### PR DESCRIPTION
**Pull request by jakobbotsch**
This adds up to 3 improvement and 3 regression example diffs to the
markdown summary for each collection. The examples are picked to be the
highest percentage wise improvements and regressions with a limit on the
total size of the diff to avoid too much text in the diffs.

### CODE REVIEW

**Comment by jakobbotsch** [comment]
Unfortunately AzDO does not render the diffs with syntax highlighting in the Extensions tab: https://dev.azure.com/dnceng-public/public/_build/results?buildId=177452&view=ms.vss-build-web.run-extensions-tab
We can color the added/removed lines manually with HTML but then the highlighting breaks for places that actually support highlighting `` ```diff `` (like GitHub).

File: src/coreclr/scripts/superpmi_diffs_summarize.py [hunk 1]
```diff
@@ -147,13 +147,86 @@ def main(main_args):
...
+        nonlocal cur_block_color
+        if cur_block is None:
+            return
+
+        style = "display:block"
+
+        if cur_block_color is not None:
+            style = style + ";background-color:" + cur_block_color
+
+        new_block_text = '<span style="{}">'.format(style) + cur_block + "</span>"
```
Comment (am11) [inline src/coreclr/scripts/superpmi_diffs_summarize.py:203]:
```suggestion
        new_block_text = '<div style="{}">'.format(style) + cur_block + "</div>"
```
before:
Screenshot 2023-02-19 at 22 49 57

after:
Screenshot 2023-02-19 at 22 51 14

Comment (jakobbotsch) [reply src/coreclr/scripts/superpmi_diffs_summarize.py:203]: The problem is that AzDO filters out block elements, it removes my `display:block` style from the spans. I have a fix where I just pad everything to the right length, but let me try to see if `div` works first or if it also filters that out.

Comment (jakobbotsch) [reply src/coreclr/scripts/superpmi_diffs_summarize.py]: Nice, the div instead does the trick. Thank you!

**Comment by jakobbotsch** [comment]
> We can color the added/removed lines manually with HTML but then the highlighting breaks for places that actually support highlighting `` ```diff `` (like GitHub).

Doing this in superpmi_diffs_summarize.py was manageable and saves us from having to pass through some --ci flag to superpmi.py. [This](https://dev.azure.com/dnceng-public/public/_build/results?buildId=177727&view=ms.vss-build-web.run-extensions-tab) is the example of the diffs page. A screenshot:
image

PTAL @dotnet/jit-contrib

**Review by BruceForstall — APPROVED** [review]
Very cool! LGTM.

**Pull request merged**